base/src/main/java/org/geysermc/api/util/ApiVersion.java

```diff
@@ -0,0 +1,91 @@
+package org.geysermc.api.util;
+
+/**
+ * Represents a version of the api.
+ */
+public class ApiVersion {
```

Comment:
Can't this be a record?

Reply:
https://github.com/GeyserMC/api/blob/master/build-logic%2Fsrc%2Fmain%2Fkotlin%2Fapi.base-conventions.gradle.kts#L10-L11
The Java target version is Java 8, so.. not unless that's changed

Reply:
Just putting this here: seems like its been decided to hold off on updating the java version.